File: src/main/java/com/king/spark/sql/GenericLoadSave.java
package com.king.spark.sql;


import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SparkSession;

/**
 * 通用的load和save操作
 *
 * @author king
 */
public class GenericLoadSave {

    public static void main(String[] args) {
        SparkSession spark = SparkSession
                .builder()
                .master("local")
                .appName("GenericLoadSave")
                .getOrCreate();

        Dataset usersDF = spark.read().load(
                "hdfs://spark1:9000/users.parquet");
        usersDF.select("name", "favorite_color").write()
                .save("hdfs://spark1:9000/namesAndFavColors.parquet");
    }

}

File: src/main/scala/com/king/spark/sql/GenericLoadSave.scala
package com.king.spark.sql

import org.apache.spark.sql.SparkSession


/**
  * @author king
  */
object GenericLoadSave {

  def main(args: Array[String]): Unit = {
    val spark = SparkSession
      .builder()
      .appName("GenericLoadSave")
      .getOrCreate()

    val usersDF = spark.read.load("hdfs://spark1:9000/users.parquet")
    usersDF.write.save("hdfs://spark1:9000/namesAndFavColors_scala")
  }

}